schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -653,9 +653,18 @@ public void deleteSchemaBranch(Long schemaBranchId) throws SchemaBranchNotFoundE
             SchemaBranchVersionMapping schemaBranchVersionMapping = schemaBranchVersionMappingIterator.next();
             Long schemaVersionId = schemaBranchVersionMapping.getSchemaVersionInfoId();
             try {
+                List<QueryParam> schemaVersionCountParam = new ArrayList<>();
+                schemaVersionCountParam.add(new QueryParam(SchemaBranchVersionMapping.SCHEMA_VERSION_INFO_ID, schemaBranchVersionMapping.getSchemaVersionInfoId().toString()));
+                Collection<SchemaBranchVersionMapping> mappingsForSchemaTiedToMutlipleBranch = storageManager.find(SchemaBranchVersionMapping.NAMESPACE, schemaVersionCountParam);
+                if (mappingsForSchemaTiedToMutlipleBranch.size() > 1) {
+                    SchemaVersionInfo schemaVersionInfo = schemaVersionLifecycleManager.getSchemaVersionInfo(new SchemaIdVersion(schemaVersionId));
+                    Long forkedBranchId = mappingsForSchemaTiedToMutlipleBranch.stream().filter(mapping -> !mapping.getSchemaBranchId().equals(schemaBranchId)).findFirst().get().getSchemaBranchId();
```

Comment:
Good to show all the schema versions with their forked branches instead of showing only one of them in error message.

Reply:
@satishd: If we have a lot of branches forked out from the schema branch to be deleted, then showing the list of the all the schema versions with the branch in the error message might become too long and the UI might get cluttered.

Reply:
@raju-saravanan Giving the appropriate information to user is important. Message can be formatted in right manner and UI should wrap that up and show respectively.